rs/consensus/idkg/src/pre_signer.rs

```diff
@@ -527,15 +575,15 @@ impl IDkgPreSignerImpl {
             .map(|transcript_params| transcript_params.transcript_id)
             .collect::<BTreeSet<_>>();
 
-        let mut ret = Vec::new();
+        let ret = std::iter::empty();
         let current_height = block_reader.tip_height();
         let mut target_subnet_xnet_transcripts = BTreeSet::new();
         for transcript_params_ref in block_reader.target_subnet_xnet_transcripts() {
             target_subnet_xnet_transcripts.insert(transcript_params_ref.transcript_id);
         }
 
         // Unvalidated dealings.
-        let mut action = idkg_pool
+        let action = idkg_pool
```

Comment:
Could we also do this?
```suggestion
        let ret = idkg_pool
```
And then drop the empty iterator above and `let ret = ret.chain(action);` below.

Reply:
https://github.com/dfinity/ic/pull/6526/commits/2a793012308530cb8fa79d3f31934b11070015f3